### ISSUE TITLE
autosubmit version in autosubmit.conf

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#441]
In GitLab by @pablohe on Oct 30, 2019, 12:51

```
[config]
...
# No need to change.
# Autosubmit version identifier
AUTOSUBMIT_VERSION = 3.12.0b0
...
```

Some users are confused about this parameter, they think that is the version that was launch the last time.

This parameter is used in any place?

I have a question related: Is the AS version used in the runs storaged in any place ? e.g.: in tmp log files?

I think this will be use full to debug

What do you think?

@dbeltrankyl @wuruchi @mcastril @ jberlin

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 30, 2019, 12:55

I assume you mean "confused".

This parameter refers to the AS version used to create the experiment. I think it doesn't have any further utility.

The version is stored also in the DDBB and can be seen in the DDBB viewers.

**Comment by kinow** [comment]
In GitLab by @ jberlin on Oct 30, 2019, 12:56

Hola
In my opinion, user should use the deployed production release of Autosubmit by default, I would remove that setting to avoid confusion unless different AS versions completely differs in terms of functionality provided

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 30, 2019, 13:01

I don't think it is totally useless. User's should use the default version for production purposes, but we always foster to test newer versions so we know if they are working or not. Otherwise we would never know when to move a development version to default.

What we maybe could change is the modification of this parameter. I think it should be re-generated on "Autosubmit create". This way the testing suite experiments would be marked with the AS version and the users should know which one is supported by the tests. In that case Autosubmit should also update the DDBB.

Do you think this could help?

**Comment by kinow** [comment]
In GitLab by @pablohe on Nov 8, 2019, 09:12

mentioned in issue auto-ecearth3#1110

**Comment by kinow** [comment]
In GitLab by @mcastril on Nov 11, 2019, 11:20

@ yruprich suggested to use this VERSION parameter as a check for Autosubmit to know which version the user is expected to use. So if the user intends to run the experiment with another version, Autosubmit should complain and not run until the user manually updates the VERSION option.

This would be an extra security layer for production exps. Would it prevent people from using development versions? In any case we should inform well what this error means.

**Comment by kinow** [comment]
In GitLab by @pablohe on Nov 13, 2019, 15:27

unassigned @pablohe

**Comment by kinow** [comment]
In GitLab by @mcastril on Nov 18, 2019, 09:21

So the change will be:

* Autosubmit run will look the value of `AUTOSUBMIT_VERSION` and it will only run if this is empty or the current version used.

**Comment by kinow** [comment]
In GitLab by @mcastril on Nov 18, 2019, 15:26

The Autosubmit version information should be also added to the Autosubmit run log

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jan 3, 2020, 16:50

mentioned in commit dcf9e066f3b8510b54a6e28836247ea543309ca6

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jan 3, 2020, 16:50

mentioned in commit 48927927d4bae1f89b14bc2ae1a8db145df527e2

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jan 7, 2020, 09:53

Hi,

implemented on master branch(3.12.0) still not on the modules ( git pipeline issue)

The behavior of run will change to adapt to these changes.

Now, autosubmit run will only run if the **AUTOSUBMIT version** and **experiment version** are equal, otherwise, it will display an error indicating how to update the experiment version. If the experiment version is **empty** the experiment should also be allowed to run.

for update the experiment version, the user can perform:

* `autosubmit run <expid> -v` 
* `autosubmit updateversion <expid>`
* `edit the parameter version on <experiment_root>/conf/autosubmit_expid_.conf`

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 20, 2020, 16:07

closed
